src/coefficients/StdFunctionCoefficient.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+// # Author: Jonathan Wong @ LLNL.
+
+#include "StdFunctionCoefficient.hpp"
+
+StdFunctionCoefficient::StdFunctionCoefficient(std::function<double(mfem::Vector &)> func) :
+  func_(func)
+{ }
+
+double StdFunctionCoefficient::Eval(ElementTransformation &T, const IntegrationPoint &ip)
+{
+  double x[T.GetSpaceDim()];
+  Vector transip(x, T.GetSpaceDim());
+
+  T.Transform(ip, transip);
+  return func_(transip);
+}
+
+StdFunctionVectorCoefficient::StdFunctionVectorCoefficient(int dim,
+    std::function<void(mfem::Vector &, mfem::Vector &)> func) :
+  VectorCoefficient(dim),
+  func_(func)
+{ }
+
+void StdFunctionVectorCoefficient::Eval(Vector &V, ElementTransformation &T, const IntegrationPoint &ip)
+{
+  double x[T.GetSpaceDim()];
```

Comment:
Variable Length Arrays were dropped off the standard in C99.  They should be avoided due to undefined behavior and some compilers will not compile them.  "-pedantic" will add the warning.  We should probably address our warning/error flags.

Please use a shared_ptr or a unique_ptr.  There is a PR that @samuelpmishLLNL  is doing to look into moving towards those.  If you don't feel comfortable with those change to new/delete and we can update it after.